### PR TITLE
drivers: counter: esp32: fix missing absolute alarm flag check

### LINKED ISSUE #50368: esp32: counter driver not working with absoulte value
**Describe the bug**
`COUNTER_ALARM_CFG_ABSOLUTE` is not working in esp32 platform.

### PR DESCRIPTION
**Pull request by sylvioalves**
Current ESP32 counter driver is not implementing the absolute alarm option. 

Closes #50368

### CODE REVIEW

**Comment by sylvioalves** [comment]
I consider this a bug as it won't work when the absolute flag is selected. In case it is a feature, no problem waiting release window to end.

**Review by LucasTambor — APPROVED** [review]

File: drivers/counter/counter_esp32_rtc.c [hunk 1]
```diff
@@ -107,8 +107,13 @@ static int counter_esp32_set_alarm(const struct device *dev, uint8_t chan_id,
 
 	counter_esp32_get_value(dev, &now);
 
+	if ((alarm_cfg->flags & COUNTER_ALARM_CFG_ABSOLUTE) == 0) {
+		WRITE_PERI_REG(RTC_CNTL_SLP_TIMER0_REG, (now + alarm_cfg->ticks));
+	} else {
+		WRITE_PERI_REG(RTC_CNTL_SLP_TIMER0_REG, alarm_cfg->ticks);
```
Comment (uLipe) [inline drivers/counter/counter_esp32_rtc.c:113]: Very nit question,are there not hal like functions equivalent to this?

**Review by uLipe — APPROVED** [review]
Nit question, not a blocker tough.

File: drivers/counter/counter_esp32_rtc.c (hunk 1 above)
Comment (sylvioalves) [reply drivers/counter/counter_esp32_rtc.c:113]: No, `hal_espressif` also uses the same approach. The only thing that is not implemented is supporting `uint64_t` for ticks value. But this requires Zephyr's API change.

**Pull request merged**